kedro/runner/thread_runner.py

```diff
@@ -49,14 +49,14 @@ def __init__(self, max_workers: int = None, is_async: bool = False):
         self._max_workers = max_workers
 
     def create_default_data_set(self, ds_name: str) -> AbstractDataSet:
-        """Factory method for creating the default data set for the runner.
+        """Factory method for creating the default dataset for the runner.
 
         Args:
-            ds_name: Name of the missing data set
+            ds_name: Name of the missing dataset.
 
         Returns:
-            An instance of an implementation of AbstractDataSet to be used
-            for all unregistered data sets.
+            An instance of ``MemoryDataSet`` to be used for all
```

Comment:
I don't think this is true, there's no requirement for it to be a `MemoryDataSet` and in fact for some runners it shouldn't be a `MemoryDataSet`.

Reply:
This is specifically on the `ThreadRunner.create_default_data_set` docstring, similar to how `ParallelRunner.create_default_data_set` already had a custom docstring mentioning `_SharedMemoryDataSet`.

Actually, I think the one oversight (in keeping consistent with `ParallelRunner`) is that the return type here should be `-> MemoryDataSet`. WDYT?

Reply:
Yeah I agree with the docstrings. I almost flagged the same thing until I saw that in the abstract class we keep it generic, and in the implementations it's concrete. I agree we should make the typehint here be `MemoryDataSet`.